### ISSUE TITLE
Can you provide a script for inference on any image?

### ISSUE THREAD

**Issue by iAmJuan550** [SysCV/qdtrack#61]
I would like to test your pre-trained models on images taken in the wild to see how well they behave

**Comment by iAmJuan550** [comment]
I solved this problem. Here is a step by step guide:

1) Run `fake_bdd100k.py`

```
import os
import json
import natsort

img_files = natsort.natsorted(os.listdir('org_imgs'))

fake_bdd100k_file = open('fake_bdd100k_file.json', 'w+')
fake_data = []
for i in range(len(img_files)):
	fake_label_dict = {"id": "00114122", "category": "car", 
	"attributes": {"Occluded": True, "Truncated": False, "Crowd": False}, 
	"box2d": {"x1": 341.5677570408605, "x2": 750.9399525709422, "y1": 276.76071308327244, "y2": 585.7208606531455}}
	fake_data.append({'name': img_files[i], 'labels': [fake_label_dict], 
	"video_name": "org_imgs", "index": i})
json.dump([fake_data], fake_bdd100k_file)
fake_bdd100k_file.close()
```
2) Go to your terminal and run the following:
`python -m bdd100k.label.to_coco -m track -i fake_bdd100k_file.json -o fake_coco.json`

3) Run `custom qdtrack.py`:

```
import torch
import os
import json
import natsort
import cv2
import numpy as np
import mmcv
from mmcv import Config
from mmcv.image import imread, tensor2imgs
from qdtrack.models import build_model
from qdtrack.datasets import build_dataloader
from mmdet.datasets import build_dataset
from mmcv.runner import get_dist_info, load_checkpoint
from mmcv.cnn import fuse_conv_bn
from mmcv.parallel import MMDataParallel
from torch.utils.data import DataLoader, Dataset

def write_stroke_texts(image, strs, locs, size_mul=1, blk_mul=1, wht_mul=1):
    font = cv2.FONT_HERSHEY_SIMPLEX
    for text_str, loc in zip(strs, locs):
        cv2.putText(image, text_str, loc, font, 0.5 * size_mul, (0, 0, 0), 2 * blk_mul, cv2.LINE_AA)
        cv2.putText(image, text_str, loc, font, 0.5 * size_mul, (255, 255, 255), 1 * wht_mul, cv2.LINE_AA)


session_name = 'TestSession-2021-07-28-12:56:56'
root_dir = os.path.join('/home/muhammadmehdi/PycharmProjects/VIDEOS', session_name)
cfg = Config.fromfile('configs/qdtrack-frcnn_r50_fpn_12e_bdd100k.py')

# set cudnn_benchmark
torch.backends.cudnn.benchmark = True
cfg.model.pretrained = None
cfg.data.test.test_mode = True

# init distributed env first, since logger depends on the dist info.
distributed = False

# build the model and load checkpoint
model = build_model(cfg.model, train_cfg=None, test_cfg=None)
checkpoint = load_checkpoint(model, 'weights/qdtrack-frcnn_r50_fpn_12e_bdd100k-13328aed.pth', map_location='cuda')

model = fuse_conv_bn(model)

if 'CLASSES' in checkpoint['meta']:
    model.CLASSES = checkpoint['meta']['CLASSES']
else:
    model.CLASSES = dataset.CLASSES

model = MMDataParallel(model, device_ids=[0])
cfg.data.test['ann_file'] = 'fake_coco.json'
cfg.data.test['img_prefix'] = root_dir

dataset = build_dataset(cfg.data.test)
data_loader = build_dataloader(
        dataset,
        samples_per_gpu=1,
        workers_per_gpu=cfg.data.workers_per_gpu,
        dist=False,
        shuffle=False)
model.eval()
for i, data in enumerate(data_loader):
    with torch.no_grad():
        result = model(return_loss=False, rescale=True, **data)
        img_tensor = data['img'][0]
        img_metas = data['img_metas'][0].data[0]
        imgs = tensor2imgs(img_tensor, **img_metas[0]['img_norm_cfg'])
        for i, (img, img_meta) in enumerate(zip(imgs, img_metas)):
            h, w, _ = img_meta['img_shape']
            img_show = img[:h, :w, :]

            ori_h, ori_w = img_meta['ori_shape'][:-1]
            print(ori_h, ori_w)
            img_show = mmcv.imresize(img_show, (ori_w, ori_h))

            img = imread(img)
            img = np.ascontiguousarray(img)

            bboxes = result['bbox_result']
            trks = result['track_result']
            if len(trks) == 0:
                continue
            if isinstance(bboxes, np.ndarray):
                bboxes = [bboxes]
            tracks = []
            for trk_id, trk_dict in trks.items():
                x1, y1, x2, y2, _ = trk_dict['bbox']
                tracks.append(np.array([x1, y1, x2, y2, trk_id, trk_dict['label']]))
            for j, (_bboxes, _tracks) in enumerate(zip(bboxes, tracks)):
                _bboxes = _bboxes.astype(np.int32)
                _tracks = _tracks.astype(np.int32)
                left_top = (_tracks[0], _tracks[1])
                x1, y1 = left_top
                right_bottom = (_tracks[2], _tracks[3])
                x2, y2 = right_bottom
                cv2.rectangle(img_show, left_top, right_bottom, (0, 255, 0), 3)
                write_stroke_texts(img_show, [str(_tracks[4]), 'COCO ID ' + str(_tracks[-1])],
                                   [(x1, y1 - 10), (x1, y1 - 20)])
            cv2.imshow('', img_show)
            cv2.waitKey(1000)
```
